src/parser/xmlHandler.js

```diff
@@ -58,23 +59,52 @@ class XMLHandler {
           return node;
         }
       }
+      if(isSimple && val !== null && typeof val === "object"){
+        if (typeof val[this.options.attributesKey] !== "undefined"){
+          attrs = val[this.options.attributesKey];
+        }
+        if (typeof val[this.options.valueKey] !== "undefined"){
+          val = val[this.options.valueKey];
+        } 
+	    }
+      
       let element;
       if (descriptor.form === 'unqualified') {
         element = isSimple ? node.element(name, val) : node.element(name);
       } else if (descriptor.qname) {
         nsContext.pushContext();
-        let mapping = declareNamespace(nsContext, null,
+	      let mapping = nsContext.getPrefix(descriptor.qname.nsURI);
```

Comment:
@raymondfeng I am adding more comments and correcting indentation. Will request for a PTAL tomorrow